presto-main/src/main/java/io/prestosql/spi/expression/ConnectorExpressionTranslator.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.spi.expression;
+
+import io.prestosql.Session;
+import io.prestosql.spi.type.Type;
+import io.prestosql.sql.planner.Symbol;
+import io.prestosql.sql.planner.TypeAnalyzer;
+import io.prestosql.sql.planner.TypeProvider;
+import io.prestosql.sql.tree.AstVisitor;
+import io.prestosql.sql.tree.Expression;
+import io.prestosql.sql.tree.NodeRef;
+import io.prestosql.sql.tree.SymbolReference;
+
+import java.util.Map;
+
+public class ConnectorExpressionTranslator
+{
+    private ConnectorExpressionTranslator()
+    {
+    }
+
+    public static Expression translate(ConnectorExpression expression, Map<String, Symbol> variableMappings)
+    {
+        return new ConnectorToSqlExpressionTranslator(variableMappings).translate(expression);
+    }
+
+    public static ConnectorExpression translate(Session session, Expression expression, TypeAnalyzer types, TypeProvider inputTypes)
+    {
+        return new SqlToConnectorExpressionTranslator(types.getTypes(session, inputTypes, expression))
+                .process(expression);
+    }
+
+    private static class ConnectorToSqlExpressionTranslator
```

Comment:
this is not going to be a some kind of visitor?

Reply:
No, a visitor would be overkill. Eventually, I hope to use advanced switch & pattern matching when that finally make it into Java, so it will be an easy conversion.